src/System.Reflection/tests/FieldInfo/FieldInfo_RTGenericTests.cs

```diff
@@ -3,97 +3,95 @@
 // See the LICENSE file in the project root for more information.
 
 using Xunit;
-using System;
-using System.Reflection;
 using System.Collections.Generic;
 
-
 #pragma warning disable 0414
 
-
 namespace System.Reflection.Tests
 {
     public class FieldInfoRTGenericTests
     {
-        //Verify FieldInfo for generic Types
-        [Fact]
-        public static void TestGenericFields()
+        public static IEnumerable<object[]> FieldInfoRTGenericTests_TestData()
         {
-            FieldInfoRTGenericTests myInstance = new FieldInfoRTGenericTests();
-            myInstance.populateScenario();
-
-            Type openType = typeof(PublicFieldGeneric<>);
-            foreach (Scenario sc in list)
-            {
-                Type type = openType.MakeGenericType(new Type[] { sc.gaType });
-                Object obj = Activator.CreateInstance(type);
-                FieldInfo fi = null;
-
-                fi = getField(type, sc.fieldName);
-                Assert.True((fi.GetValue(obj)).Equals(sc.initialValue), "Get Value should return value that was set for generic type: " + sc.fieldName);
-
-                fi.SetValue(obj, sc.changedValue);
-                Assert.True((fi.GetValue(obj)).Equals(sc.changedValue), "Get Value should return value that was set for generic type: " + sc.fieldName);
-
-                fi.SetValue(obj, null);
-                Assert.True((fi.GetValue(obj)).Equals(sc.initialValue), "Get Value should return value that was set for generic type: " + sc.fieldName);
-            }
+            yield return new object[] { typeof(PublicFieldGeneric<>), new List<Scenario>() {
```

Comment:
Just for comparison, this is an example where MemberData does make sense, as this state can't be encoded into attributes.